mti_nma/steps/avgshape/avgshape_utils.py

```diff
@@ -42,3 +44,68 @@ def run_shcoeffs_analysis(df, savedir):
             str(savedir / Path(f'bar-{id_plot}.svg'))
         )
         plt.close(fig)
+
+
+def get_vtk_verts_faces(polydata):
+    """
+    Extracts the mesh vertices and faces encoded in the VTK polydata object
+
+    Parameters
+    ----------
+    polydata: VTK polydata object
+        Mesh extracted from .vtk file 
+    Returns
+    -------
+    mesh_verts: Numpy 2D array
+        An array of arrays giving the positions of all mesh vertices
+        mesh_verts[i][j] gives position of the ith vertex, in the jth spatial dimension
+    vmesh_faces: Numpy 2D array
+        An array of arrays listing which vertices are connected to form each mesh faces
+        mesh_faces[i] gives an array of 3 ints, indicating the indices of the
+        mesh_verts indices of vertices connected to form this trimesh faces
+    """
+
+    def get_faces(i, polydata):
+        """
+        Extracts vertices making up the ith face of the polydata mesh
+
+         Parameters
+        ----------
+        i : int
+            Index of the face for which we want to get the vertex points
+        polydata: VTK polydata object
+            Mesh extracted from .vtk file 
+        Returns
+        -------
+        mesh_verts: Numpy 1D array
+            An array of indices of vertices making up the three vertices of this face
+        """
+
+        cell = polydata.GetCell(i)
+        ids = cell.GetPointIds()
+        return np.array([ids.GetId(j) for j in range(ids.GetNumberOfIds())])
+
+    # Collect all faces and vertices into arrays
+    mesh_faces = np.array(
+        [get_faces(i, polydata) for i in range(polydata.GetNumberOfCells())]
+    )
+    mesh_verts = np.array(
```

Comment:
So this should be a numpy array of numpy arrays at the end of this operation?

Reply:
It may be better to do a list of lists and then do `np.from_array` as I think numpy may think you have an outer array of `numpy.types.object` and an inner array as `numpy.float64` or w.e. it is rather than a 2D `numpy.float64`. If that makes sense?